parser-typechecker/src/Unison/Codebase/SqliteCodebase/Migrations/MigrateSchema1To2.hs

```diff
@@ -149,6 +150,23 @@ migrateSchema1To2 getDeclType termBuffer declBuffer = do
           allDone = lift $ log $ "\nFinished migrating, initiating cleanup."
        in Sync.Progress {need, done, error = errorHandler, allDone}
 
+expectNamespaceRoot :: Sqlite.Transaction CausalHashId
+expectNamespaceRoot =
+  Sqlite.queryOneCol loadNamespaceRootSql
+
+loadNamespaceRootSql :: Sqlite.Sql
+loadNamespaceRootSql =
+  [Sqlite.sql|
+    SELECT causal_id
+    FROM namespace_root
+  |]
+
+setNamespaceRoot :: CausalHashId -> Sqlite.Transaction ()
+setNamespaceRoot id =
+  Sqlite.queryOneCol [Sqlite.sql| SELECT EXISTS (SELECT 1 FROM namespace_root) |] >>= \case
+    False -> Sqlite.execute [Sqlite.sql| INSERT INTO namespace_root VALUES (:id) |]
+    True -> Sqlite.execute [Sqlite.sql| UPDATE namespace_root SET causal_id = :id |]
+
```

Comment:
This is an ancient migration at this point, I just copied these combinators into the migration itself so I can delete them from the DB module to avoid exposing things that you really shouldn't use.